contracts/infrastructure/dapp/paraswap/ZeroExV2Filter.sol

```diff
@@ -0,0 +1,58 @@
+// Copyright (C) 2021  Argent Labs Ltd. <https://argent.xyz>
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.8.3;
+
+import "../BaseFilter.sol";
+import "./ParaswapUtils.sol";
+
+contract ZeroExV2Filter is BaseFilter {
```

Comment:
Maybe we should add a reference to Paraswap in the name of the filter to make it explicit that this is not a generic ZeroX filter (in case we decide to add a generic ZeorX filter in the future). E.g. ParaswapPoolZeroxV2Filter?

Reply:
It is actually a generic filter with a whitelist of market makers. It could be used outside of Paraswap, to trade on ZeroEx with approved market makers.

Reply:
The fact that we whitelist market makers is what makes it non generic, but I agree that we may want to add market makers outside Paraswap pools in the future. Maybe `WhitelistedZeroxV2Filter` ?